aea/protocols/dialogue/base.py

```diff
@@ -1212,26 +1313,34 @@ def teardown(self) -> None:
 class Dialogues:
     """The dialogues class keeps track of all dialogues for an agent."""
 
+    _keep_terminal_state_dialogues = False
+
     def __init__(
         self,
         self_address: Address,
         end_states: FrozenSet[Dialogue.EndState],
         message_class: Type[Message],
         dialogue_class: Type[Dialogue],
         role_from_first_message: Callable[[Message, Address], Dialogue.Role],
+        keep_terminal_state_dialogues: Optional[bool] = None,
     ) -> None:
         """
         Initialize dialogues.
 
         :param self_address: the address of the entity for whom dialogues are maintained
         :param end_states: the list of dialogue endstates
+        :param keep_terminal_state_dialogues: specify do dialogues in terminal state should stay or not
+
         :return: None
         """
 
         self._dialogues_storage = PersistDialoguesStorage(self)
         self._self_address = self_address
         self._dialogue_stats = DialogueStats(end_states)
 
+        if keep_terminal_state_dialogues is not None:
+            self._keep_terminal_state_dialogues = keep_terminal_state_dialogues
+
```

Comment:
else?

Reply:
It's updating the class variable which is set by default